tests/baselines/reference/privateNameFieldDestructuredBinding(target=es5).js

```diff
@@ -0,0 +1,62 @@
+//// [privateNameFieldDestructuredBinding.ts]
+class A {
+    #field = 1;
+    otherObject = new A();
+    testObject() {
+        return { x: 10, y: 6 };
+    }
+    testArray() {
+        return [10, 11];
+    }
+    constructor() {
+        let y: number;
+        ({ x: this.#field, y } = this.testObject());
+        ([this.#field, y] = this.testArray());
+        ({ a: this.#field, b: [this.#field] } = { a: 1, b: [2] });
+        [this.#field, [this.#field]] = [1, [2]];
+        ({ a: this.#field = 1, b: [this.#field = 1] } = { b: [] });
+        [this.#field = 2] = [];
+        [this.otherObject.#field = 2] = [];
+    }
+    static test(_a: A) {
+        [_a.#field] = [2];
+    }
+}
+
+
+//// [privateNameFieldDestructuredBinding.js]
+var __classPrivateFieldSet = (this && this.__classPrivateFieldSet) || function (receiver, privateMap, value) {
+    if (!privateMap.has(receiver)) {
+        throw new TypeError("attempted to set private field on non-instance");
+    }
+    privateMap.set(receiver, value);
+    return value;
+};
+var _field;
+var A = /** @class */ (function () {
+    function A() {
+        var _b, _c, _d, _e, _f, _g, _h, _j, _k, _l, _m;
+        var _o, _p, _q, _r, _s, _t, _u, _v, _w, _x;
+        _field.set(this, 1);
+        this.otherObject = new A();
+        var y;
+        (_o = this, (_b = this.testObject(), ({ set value(_b) { __classPrivateFieldSet(_o, _field, _b); } }).value = _b.x, y = _b.y, _b));

```

Comment:
We may need a heuristic to detect and unroll `({ set value(_b) { __classPrivateFieldSet(_o, _field, _b); } }).value = ` when targeting ES5/3. The setter wrapper isn't needed in ES5 and setters are not supported in ES3. Either that, or we should report an error when destructuring into a private field in ES3.

Either way, I would add `es3` to the `// @target:` option for this file.

Reply:
We should probably also have an error like `Private named fields are only available when targeting ECMAScript 2015 and higher.` emitted on any private named declaration, right?